developer/src/kmc-model/src/build-trie.ts

```diff
@@ -291,196 +213,25 @@ namespace Trie {
-    } else {
-      // Push the node down to a lower node.
-      addItemToInternalNode(node, entry, index);
-    }
-
-    node.unsorted = true;
-  }
-
-  /**
-   * Adds an item to the internal node at a given depth.
-   * @param item
-   * @param index
-   */
-  function addItemToInternalNode(node: InternalNode, item: Entry, index: number) {
-    let char = item.key[index];
-    // If an internal node is the proper site for item, it belongs under the
-    // corresponding (sentinel, internal-use) child node signifying this.
-    if(char == undefined) {
-      char = INTERNAL_VALUE;
-    }
-    if (!node.children[char]) {
-      node.children[char] = createRootNode();
-      node.values.push(char);
+      totalWeight: collater.getTotalWeight(),
+      root: collater.getRoot()
     }
-    addUnsorted(node.children[char], item, index + 1);
   }
 
-  function addItemToLeaf(leaf: Leaf, item: Entry) {
-    leaf.entries.push(item);
-  }
-
-  /**
-   * Mutates the given Leaf to turn it into an InternalNode.
-   *
-   * NOTE: the node passed in will be DESTRUCTIVELY CHANGED into a different
-   * type when passed into this function!
-   *
-   * @param depth depth of the trie at this level.
-   */
-  function convertLeafToInternalNode(leaf: Leaf, depth: number): void {
-    let entries = leaf.entries;
-
-    // Alias the current node, as the desired type.
-    let internal = (<unknown> leaf) as InternalNode;
-    internal.type = 'internal';
-
-    delete leaf.entries;
-    internal.values = [];
-    internal.children = {};
-
-    // Convert the old values array into the format for interior nodes.
-    for (let item of entries) {
-      let char: string;
-      if (depth < item.key.length) {
-        char = item.key[depth];
-      } else {
-        char = INTERNAL_VALUE;
-      }
-
-      if (!internal.children[char]) {
-        internal.children[char] = createRootNode();
-        internal.values.push(char);
-      }
-      addUnsorted(internal.children[char], item, depth + 1);
-    }
-
-    internal.unsorted = true;
-  }
-
-  /**
-   * Recursively sort the trie, in descending order of weight.
-   * @param node any node in the trie
-   */
-  function sortTrie(node: Node) {
-    if (node.type === 'leaf') {
-      if (!node.unsorted) {
-        return;
-      }
-
-      node.entries.sort(function (a, b) { return b.weight - a.weight; });
-    } else {
-      // We MUST recurse and sort children before returning.
-      for (let char of node.values) {
-        sortTrie(node.children[char]);
-      }
-
-      if (!node.unsorted) {
-        return;
-      }
-
-      node.values.sort((a, b) => {
-        return node.children[b].weight - node.children[a].weight;
-      });
-    }
-
-    delete node.unsorted;
-  }
-
-  /**
-   * O(n) recursive traversal to sum the total weight of all leaves in the
-   * trie, starting at the provided node.
-   *
-   * @param node The node to start summing weights.
-   */
-  function sumWeights(node: Node): number {
```

Comment:
This method is entirely replaced within the new version with something simple yet more efficient; other comments document this.